components/org.wso2.carbon.identity.application.authenticator.facebook/src/main/java/org/wso2/carbon/identity/application/authenticator/facebook/FacebookAuthenticator.java

```diff
@@ -366,13 +368,38 @@ protected void buildClaims(AuthenticationContext context, Map<String, Object> js
                 }
             }
 
+            if (StringUtils.isNotEmpty(claimDialectUri)) {
+                claimDialectUri = claimDialectUri.substring(0, claimDialectUri.length() - 1);
+            }
+            boolean isUserClaimURIEmpty = false;
             if (StringUtils.isBlank(context.getExternalIdP().getIdentityProvider()
                     .getClaimConfig().getUserClaimURI())) {
-                context.getExternalIdP().getIdentityProvider().getClaimConfig().setUserClaimURI
-                        (FacebookAuthenticatorConstants.EMAIL);
+                isUserClaimURIEmpty = true;
```

Comment:
Let's move this variable to top of the method

Reply:
done